### ISSUE TITLE
Add additional fields in many2many relation

### ISSUE THREAD

**Issue by kirtangajjar** [odoo/odoo#37856]
Impacted versions: All

Current behavior:

We cannot add additional fields in many2many relation. The transaction table which holds many2many relation, has only 2 columns to track id of two objects. We should be able to add more columns to them.

This feature is [very useful](https://www.odoo.com/forum/help-1/question/create-extra-field-on-many2many-relationship-29747) and might help Odoo core itself. One place I could see it being useful is in `project` module where the sequence of `task stage` is defined on the model itself. So if there are multiple `projects` referencing same `stage`, changing sequence of `stage` in one `project` changes it for all `projects`.

Expected behavior:

If we had the ability to add stage sequence on the intermediate table, then we could have referenced same task stage in multiple projects using different sequence. This is just an example, but this feature will be useful in many cases.

**Comment by pedrobaeza** [comment]
Sorry, you have to use one2many for that.

Closing as no issue, but misuse problem.

**Comment by matuqam** [comment]
This worked for me on OERP 7. The situation was two existing models (A, B) linked by a many2many relation. I Created a new model (C) linked to A with a many2one and linked to B with a many2one. I used the same field names as the ones used by models A and B for the many2one fields in C. I gave C a _table attribute with the value corresponding to the name of the existing many2many table linking A and B. Possibly this verbose explanation will detail what is implied by pedrobaeza's short, yet accurate anwser, for those who are more novice.
